test/completions2.jl

```diff
@@ -0,0 +1,58 @@
+module completions2
```

Comment:
I added this test to exercise the whole completion pipeline but it feels like this is far too complex, and it’s exactly the kind of thing that would likely break in the future e.g. when LS were made async, so I feel it could become quickly technical debt down the road. 

For now, I think I will have this test, but it might be better to factor out the part that creates the file cache so we can easily write proper unit tests.